src/app/app.component.ts

```diff
@@ -27,9 +27,20 @@ export class AppComponent implements OnInit {
     this.count$ = this.store.select(getCount);
     this.todos$ = this.store.select(getTodos);
 
-    this.currentTodo$ = this.todos$
-      .map(todos => todos.filter(todo =>
-        todo.id == this.currentTodoId)[0]);
+    /**
+     * Manual less preferred method
+     */
+    // this.selectedTodo$ = this.todos$
+    //   .combineLatest(this.selectedTodoId$)
+    //   .map(([todos, selectedTodoId]) =>
+    //     todos.filter(todo => todo.id === selectedTodoId)[0]);
```

Comment:
While we could go the route of manually selecting the "selectedTodoId" from the store and combining it with the other todos stream and then map/filter out the specific todo we want, it's preferred to have a selector instead do this combining/filtering for us.

Ultimately the goal of any redux application will be that most of your actual application logic should end up in your reducers and selectors